internal/cluster/cluster_operations.go

```diff
@@ -454,15 +457,50 @@ func getFlattenedHostSpecsForRefs(ctx context.Context, hostRefs []*models.HostRe
 	return flattenedHostSpecs, nil
 }
 
-func getFlattenedVdsSpecsForRefs(vdsSpecs []*models.VdsSpec) []map[string]interface{} {
-	flattenedVdsSpecs := *new([]map[string]interface{})
-	// Since backend API returns objects in random order sort VDSSpec list to ensure
-	// import is reproducible
-	sort.SliceStable(vdsSpecs, func(i, j int) bool {
-		return *vdsSpecs[i].Name < *vdsSpecs[j].Name
-	})
-	for _, vdsSpec := range vdsSpecs {
-		flattenedVdsSpecs = append(flattenedVdsSpecs, network.FlattenVdsSpec(vdsSpec))
+func getFlattenedVdsSpecsForRefs(ctx context.Context, clusterId string, apiClient *client.VcfClient) ([]map[string]interface{}, error) {
+	// Fetch VDS information
+	vdsParams := &clusters.GetVdsesParams{
+		ClusterID: clusterId,
+		Context:   ctx,
+	}
+	vdsResponse, err := apiClient.Clusters.GetVdses(vdsParams)
+	if err != nil {
+		return nil, err
+	}
+
+	if vdsResponse.Payload == nil {
+		log.Fatal("vdsResponse.Payload is nil")
 	}
-	return flattenedVdsSpecs
+
+	flattenedVdsSpecs := make([]map[string]interface{}, len(vdsResponse.Payload))
+	for i, vds := range vdsResponse.Payload {
+		portGroupSpecs := make([]*models.PortgroupSpec, len(vds.PortGroups))
+		for j, pg := range vds.PortGroups {
+			portGroupSpecs[j] = &models.PortgroupSpec{
+				Name:          pg.Name,
+				TransportType: pg.TransportType,
+			}
+		}
+
+		//for _, nioc := range vds.NiocBandwidthAllocations {
+		//	if nioc != nil {
+		//		fmt.Printf("NiocBandwidthAllocation: {Type: %s, shares_Allocation: %d, shares_leve: %s}\n",
+		//			nioc.Type, nioc.NiocTrafficResourceAllocation.SharesInfo.Shares, nioc.NiocTrafficResourceAllocation.SharesInfo.Level)
+		//		fmt.Println("Type:", reflect.TypeOf(nioc.Type))
+		//		fmt.Println("Shares:", reflect.TypeOf(nioc.NiocTrafficResourceAllocation.SharesInfo.Shares))
+		//		fmt.Println("Level:", reflect.TypeOf(nioc.NiocTrafficResourceAllocation.SharesInfo.Level))
+		//	} else {
+		//		fmt.Println("NiocBandwidthAllocation: <nil>")
+		//	}
+		//}
```

Comment:
Try this:

```suggestion
		niocSpecs := make([]*models.NiocBandwidthAllocationSpec, len(vds.NiocBandwidthAllocations))
		for k, nioc := range vds.NiocBandwidthAllocations {
			if nioc != nil {
				niocSpecs[k] = &models.NiocBandwidthAllocationSpec{
					Type: nioc.Type,
					NiocTrafficResourceAllocation: &models.NiocTrafficResourceAllocation{
						SharesInfo: &models.SharesInfo{
							Shares: nioc.NiocTrafficResourceAllocation.SharesInfo.Shares,
							Level:  nioc.NiocTrafficResourceAllocation.SharesInfo.Level,
						},
					},
				}
			}
		}
```

Reply:
yea  i tried this code and remove it a couple times, i will try again

Reply:
Lemme know. If it doesn't I'll take a look at it.

Reply:
Any luck, @burnsjared0415?

Reply:
just looking again, just need to figure out a couple things